clusterloader2/pkg/measurement/common/etcd_metrics.go

```diff
@@ -139,13 +147,21 @@ func (e *etcdMetricsMeasurement) stopAndSummarize(host, provider string) error {
 	for _, sample := range samples {
 		switch sample.Metric[model.MetricNameLabel] {
 		case "etcd_disk_backend_commit_duration_seconds_bucket":
+			e.Lock()
```

Comment:
please avoid locking in every case, just do lock before switch and unlock at the end

or even better (which you could do also above), is to opaque the critical section into func and use defer, i.e.
```
func () {
  e.Lock()
  defer e.Unlock()
  switch sample....
}()
```

What you get then is that even if you add return at some point or sth like that, it will still work.

Reply:
Thank you very much for your comment @wojtek-t !
Fixed it